src/plugins/mvcc_delete_plugin.cpp

```diff
@@ -0,0 +1,158 @@
+        const CommitID max_commit_id_diff =
+            table->max_chunk_size() * static_cast<CommitID>(_DELETE_THRESHOLD_COMMIT_DIFF_FACTOR);
+
+        // Evaluate metrics
+        const bool criterion1 = _DELETE_THRESHOLD_RATE_INVALIDATED_ROWS <= invalidated_rows_ratio;
+        const bool criterion2 = max_commit_id_diff <= commit_id_diff;
+
+        if (criterion1 && criterion2) {
+          const bool success = _delete_chunk_logically(table_name, chunk_id);
+
+          if (success) {
+            DebugAssert(StorageManager::get().get_table(table_name)->get_chunk(chunk_id)->get_cleanup_commit_id() !=
+                            MvccData::MAX_COMMIT_ID,
+                        "Chunk needs to be deleted logically before deleting it physically.");
+
+            std::unique_lock<std::mutex> lock(_mutex_physical_delete_queue);
+            _physical_delete_queue.emplace(table_name, chunk_id);
+          } else {
+            std::cout << "Logical delete of chunk " << chunk_id << " failed because of MVCC conflict. Retrying..."
+                      << std::endl;
+          }
+        }
+      }
+    }  // for each chunk
+  }    // for each table
+}
+
+/**
+ * This function processes the physical-delete-queue until its empty.
+ */
+void MvccDeletePlugin::_physical_delete_loop() {
+  std::unique_lock<std::mutex> lock(_mutex_physical_delete_queue);
+
+  ChunkSpecifier chunk_spec = _physical_delete_queue.front();
+  const auto& table = StorageManager::get().get_table(chunk_spec.table_name);
+  const auto& chunk = table->get_chunk(chunk_spec.chunk_id);
+
+  DebugAssert(chunk != nullptr, "Chunk does not exist. Physical Delete can not be applied.");
+
+  // Check whether there are still active transactions that might use the chunk
+  CommitID cleanup_commit_id = chunk->get_cleanup_commit_id();
+  CommitID lowest_snapshot_commit_id = TransactionManager::get().get_lowest_active_snapshot_commit_id();
+
+  if (cleanup_commit_id < lowest_snapshot_commit_id) {
+    _delete_chunk_physically(chunk_spec.table_name, chunk_spec.chunk_id);
+    _physical_delete_queue.pop();
+  }
+}
+
+bool MvccDeletePlugin::_delete_chunk_logically(const std::string& table_name, const ChunkID chunk_id) {
+  const auto& table = StorageManager::get().get_table(table_name);
+  const auto& chunk = table->get_chunk(chunk_id);
+
+  DebugAssert(chunk != nullptr, "Chunk does not exist. Physical Delete can not be applied.");
+  DebugAssert(chunk_id < (table->chunk_count() - 1),
+              "MVCC Logical Delete should not be applied on the last/current mutable chunk.");
+
+  // Create temporary referencing table that contains the given chunk only
+  auto transaction_context = TransactionManager::get().new_transaction_context();
+  auto gt = std::make_shared<GetTable>(table_name);
+  gt->set_transaction_context(transaction_context);
+
+  std::vector<ChunkID> excluded_chunk_ids(table->chunk_count());
```

Comment:
what are these three lines doing? They look like they convolutedly construct a vector that contains all ChunkID's but `chunk_id`? Can you doc this? Also be aware that iota and std::remove implies you're iterating over the ChunkID range twice, whereas a simple for loop could do what you want to do in be a single iteration.